src/main/java/com/atlassian/db/replica/api/context/QueryContext.java

```diff
@@ -0,0 +1,87 @@
+package com.atlassian.db.replica.api.context;
+
+
+import java.util.Objects;
+
+public final class QueryContext {
```

Comment:
"Context" is mega vague. Shall it hold *the* context of a query? But context is in the eye of the beholder. For some, "the" context includes thread name, for some client IP, for another it's the dialect or maybe SQL hints or a `ThreadLocal` `ReplicaConsistency` override or a bajillion different things. These things will be attracted to this class.

But we don't need all of that. What we *use* this for is to know *why* `db-replica` decided to route to main or replica. So maybe:
```suggestion
public final class RouteDecision {
```